agent/handlers/task_server_setup_sudo_integ_test.go

```diff
@@ -0,0 +1,427 @@
+		},
+		{
+			name:          "network latency and packet loss different type",
+			faultType1:    faulttype.LatencyFaultType,
+			faultType2:    faulttype.PacketLossFaultType,
+			body1:         happyNetworkLatencyReqBody,
+			body2:         happyNetworkPacketLossReqBody,
+			responseCode1: http.StatusOK,
+			responseCode2: http.StatusConflict,
+		},
+		{
+			name:          "network black hole port same type",
+			faultType1:    faulttype.BlackHolePortFaultType,
+			faultType2:    faulttype.BlackHolePortFaultType,
+			body1:         happyBlackHolePortReqBody,
+			body2:         happyBlackHolePortReqBody,
+			responseCode1: http.StatusOK,
+			responseCode2: http.StatusOK,
+		},
+		{
+			name:          "network black hole port and latency different type",
+			faultType1:    faulttype.BlackHolePortFaultType,
+			faultType2:    faulttype.LatencyFaultType,
+			body1:         happyBlackHolePortReqBody,
+			body2:         happyNetworkLatencyReqBody,
+			responseCode1: http.StatusOK,
+			responseCode2: http.StatusOK,
+		},
+		{
+			name:          "network black hole port and packet loss different type",
+			faultType1:    faulttype.BlackHolePortFaultType,
+			faultType2:    faulttype.PacketLossFaultType,
+			body1:         happyBlackHolePortReqBody,
+			body2:         happyNetworkPacketLossReqBody,
+			responseCode1: http.StatusOK,
+			responseCode2: http.StatusOK,
+		},
+	}
+
+	skipForUnsupportedTc(t)
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			// Starting a server which will include the network fault handlers
+			server, port := startServer(t)
+
+			client := &http.Client{}
+
+			// Making request 1
+			bodyBytes, _ := json.Marshal(tc.body1)
+			bodyReader := bytes.NewReader(bodyBytes)
+			req1, err := http.NewRequest(http.MethodPost, getURL(port, fmt.Sprintf(startEndpoint, endpointId, tc.faultType1)), bodyReader)
+			require.NoError(t, err)
+
+			res1, err := client.Do(req1)
+			require.NoError(t, err)
+
+			// Making request 2
+			var resp2Code int
+			// Adding a retry mechanism since there's a rate limit of only one request per 5 seconds for each network fault endpoints
+			// This is in the case of making the same requests type back to back
+			for i := 0; i < requestMaxRetries; i++ {
+				bodyBytes, _ = json.Marshal(tc.body2)
+				bodyReader = bytes.NewReader(bodyBytes)
+				req2, err := http.NewRequest(http.MethodPost, getURL(port, fmt.Sprintf(startEndpoint, endpointId, tc.faultType2)), bodyReader)
+				require.NoError(t, err)
+				res2, err := client.Do(req2)
+				require.NoError(t, err)
+				resp2Code = res2.StatusCode
+
+				if res2.StatusCode != http.StatusTooManyRequests {
+					break
+				}
+				t.Log("Retrying request...")
+				time.Sleep(requestTimeoutInSeconds)
+			}
+
+			// Cleaning up network black hole port fault injections
+			if tc.faultType1 == faulttype.BlackHolePortFaultType {
+				cleanupBlackHolePortFault(t, port, tc.body1)
+			}
+			if tc.faultType2 == faulttype.BlackHolePortFaultType {
```

Comment:
nit: cleanup is idempotent so we can make the cleanup calls without the `if` blocks?

Reply:
Also how about performing cleanup in a `defer` call so that it is always called even if the test panics?

Reply:
hmm this is encapsulated within a for loop and I don't think we'd want to do a defer here

Reply:
nvm, I will make it a defer as this is in it's own function

Reply:
done